### PR TITLE
AI Techwriter documentation update - 2024-11-28

### PR DESCRIPTION
**Pull request by PedroMiolaSilva**
- feat: adding new content to 3 sources
- refactor: making minor text changes in words and punctuation

### CODE REVIEW

File: src/content/docs/en/pages/devtools/api-graphql/queries/queries.mdx [hunk 1]
```diff
@@ -95,16 +95,26 @@ And the response to the query:
 }
 ```
 
-To query raw data, it's *mandatory* to inform:
+To perform a NOT LIKE query, you can use the `Ne` operator combined with the `Like` or `Ilike` operators to filter out entries that match a certain pattern. Here's an example:
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/devtools/api-graphql/queries/queries.mdx:98]:
```suggestion
To perform a *NOT LIKE* query, you can use the `Ne` operator combined with the `Like` or `Ilike` operators to filter out entries that match a certain pattern. Here's an example:
```

File: src/content/docs/en/pages/guides/edge-application/ea-enforce-hls.mdx [hunk 2]
```diff
@@ -126,6 +127,16 @@ Still on the **Cache Settings** tab:
...
+### Utilizing Azion Edge Storage for HLS Video Delivery
+
+To deliver MP4 videos using HLS and Azion Edge Storage:
+
+1. Upload your MP4 video files to Azion Edge Storage.
+2. Configure your encoder to convert MP4 files to HLS format and point to Azion Edge Storage as the origin.
+3. Follow the cache policy setup steps outlined above to ensure efficient video delivery.
+
+[TECHWRITER TODO: INSERT INFORMATION ABOUT HOW TO USING AZION EDGE STORAGE SPECIFICALLY FOR HLS]
+
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/guides/edge-application/ea-enforce-hls.mdx:139]:
```suggestion
```

File: src/content/docs/en/pages/main-menu/reference/secure/edge-firewall/web-application-firewall/custom-allowed-rule.mdx [hunk 3]
```diff
@@ -92,7 +92,16 @@ See the list of all available internal rules below:
 | 1402    | Possible trick to evade protection: encoded chars `%20-%3F` found in Body, Path, Query String or Cookies.                                  |
 | 1500    | Possible File Upload attempt: `asp/php` or `.ph`, `.asp`, `.ht` found in filename in a multipart POST containing a file.                   |
 
-<iframe width="560" height="315" src="https://www.youtube.com/embed/uhpoeBXLvxk?si=eb8Jx_LRaUq1FBlh" loading="lazy" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
+## Testing Rules in Staging Environment
+
+To ensure that your custom allowed rules are working as intended, it is recommended to test them in a staging environment before deploying them to production. A staging environment allows you to simulate real-world traffic and catch potential issues without affecting your live application.
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/main-menu/reference/secure/edge-firewall/web-application-firewall/custom-allowed-rule.mdx:97]:
```suggestion
To ensure that your custom-allowed rules are working as intended, it's recommended that you test them in a staging environment before deploying them to production. A staging environment allows you to simulate real-world traffic and catch potential issues without affecting your live application.
```

File: src/content/docs/en/pages/main-menu/reference/secure/edge-firewall/web-application-firewall/custom-allowed-rule.mdx [hunk 4]
```diff
@@ -92,7 +92,16 @@ See the list of all available internal rules below:
...
 
+1. **Set Up a Staging Environment:** Create a separate instance of your application that mirrors your production environment as closely as possible.
+2. **Deploy Rules:** Implement your custom allowed rules in this staging environment.
+3. **Simulate Traffic:** Generate traffic patterns that resemble expected user behavior as well as potential threat scenarios.
+4. **Monitor Logs and Alerts:** Keep an eye on the logs to verify that legitimate requests are allowed and false positives are minimized.
+5. **Adjust and Optimize:** Based on the results, fine-tune your rules to improve accuracy and effectiveness.
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/main-menu/reference/secure/edge-firewall/web-application-firewall/custom-allowed-rule.mdx:103]:
```suggestion
1. **Set up a staging environment**: create a separate instance of your application that mirrors your production environment as closely as possible.
2. **Deploy rules**: implement your Custom Allowed Rules in this staging environment.
3. **Simulate traffic**: generate traffic patterns that resemble expected user behavior and potential threat scenarios.
4. **Monitor logs and alerts**: check the logs to verify that legitimate requests are allowed and false positives are minimized.
5. **Adjust and optimize**: based on the results, fine-tune your rules to improve accuracy and effectiveness.
```

File: src/content/docs/en/pages/main-menu/reference/secure/edge-firewall/web-application-firewall/custom-allowed-rule.mdx [hunk 5]
```diff
@@ -92,7 +92,16 @@ See the list of all available internal rules below:
...
 
+By following these steps, you can ensure that your WAF custom allowed rules provide the desired security enhancements without unintended consequences.
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/main-menu/reference/secure/edge-firewall/web-application-firewall/custom-allowed-rule.mdx:105]:
```suggestion
By following these steps, you can ensure that your WAF Custom Allowed Rules provide the desired security enhancements without unintended consequences.
```

File: src/content/docs/en/pages/guides/edge-application/ea-enforce-hls.mdx [hunk 6]
```diff
@@ -126,6 +127,14 @@ Still on the **Cache Settings** tab:
 Read more about setting up [cache settings](/en/documentation/products/build/edge-application/cache-settings/) for your edge applications.
 :::
 
+### Utilizing Azion Edge Storage for HLS Video Delivery
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/guides/edge-application/ea-enforce-hls.mdx:130]:
```suggestion
### Utilizing Azion Edge Storage for HLS video delivery
```

File: src/content/docs/en/pages/main-menu/reference/secure/edge-firewall/web-application-firewall/custom-allowed-rule.mdx [hunk 7]
```diff
@@ -92,7 +92,16 @@ See the list of all available internal rules below:
 | 1402    | Possible trick to evade protection: encoded chars `%20-%3F` found in Body, Path, Query String or Cookies.                                  |
 | 1500    | Possible File Upload attempt: `asp/php` or `.ph`, `.asp`, `.ht` found in filename in a multipart POST containing a file.                   |
 
-<iframe width="560" height="315" src="https://www.youtube.com/embed/uhpoeBXLvxk?si=eb8Jx_LRaUq1FBlh" loading="lazy" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
+## Testing Rules in Staging Environment
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/main-menu/reference/secure/edge-firewall/web-application-firewall/custom-allowed-rule.mdx:95]:
```suggestion
## Testing Rules in the staging environment
```

File: src/content/docs/en/pages/main-menu/reference/secure/edge-firewall/web-application-firewall/custom-allowed-rule.mdx [hunk 8]
```diff
@@ -92,7 +92,16 @@ See the list of all available internal rules below:
...
 
+1. **Set up a staging environment**: create a separate instance of your application that mirrors your production environment as closely as possible.
+2. **Deploy rules**: implement your Custom Allowed Rules in this staging environment.
+3. **Simulate traffic**: generate traffic patterns that resemble expected user behavior and potential threat scenarios.
+4. **Monitor logs and alerts**: check the logs to verify that legitimate requests are allowed and false positives are minimized.
+5. **Adjust and optimize**: based on the results, fine-tune your rules to improve accuracy and effectiveness.
 
+By following these steps, you can ensure that your WAF Custom Allowed Rules provide the desired security enhancements without unintended consequences.
 
+<iframe width="560" height="315" src="https://www.youtube.com/embed/uhpoeBXLvxk?si=eb8Jx_LRaUq1FBlh" loading="lazy" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/main-menu/reference/secure/edge-firewall/web-application-firewall/custom-allowed-rule.mdx:107]:
```suggestion
<iframe width="560" height="315" src="https://www.youtube.com/embed/uhpoeBXLvxk?si=eb8Jx_LRaUq1FBlh" loading="lazy" title="Custom allowed rules Web Application Firewall" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
```

File: src/content/docs/en/pages/devtools/api-graphql/queries/queries.mdx [hunk 9]
```diff
@@ -380,9 +401,7 @@ Depending on the [type of field of a dataset](/en/documentation/devtools/graphql
 | String | `Eq`, `Ne`, `Like`, `Ilike`, `In`, `NotIn`, `IsNull`  |
 | Integer, Float, DateTime | `Eq`, `Lt`, `Lte`, `Gt`, `Gte`, `Ne`, `In`, `NotIn`, `IsNull`, `Range` |
 
-
 Discover the essential role of operators in GraphQL queries. Watch the video below:
 <div class="cms-embed">
   <iframe width="600" height="400" src="https://www.youtube.com/embed/OGam-xrzuBA?si=cg-SIvWH8OedkrtY" loading="lazy" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/devtools/api-graphql/queries/queries.mdx:406]:
```suggestion
  <iframe width="600" height="400" src="https://www.youtube.com/embed/OGam-xrzuBA?si=cg-SIvWH8OedkrtY" loading="lazy" title="Operators in GraphQL for Real Time Event Queries" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
```

File: src/content/docs/en/pages/guides/edge-application/ea-enforce-hls.mdx [hunk 10]
```diff
@@ -23,6 +23,7 @@ Azion Edge Platform allows you to [deliver live streaming content in HLS format]
 - An existing Azion edge application, or [create a new one](/en/documentation/products/guides/build/build-an-application/).
 - [A domain associated with the edge application](/en/documentation/products/guides/configure-a-domain/).
 - [Tiered Cache enabled](/en/documentation/products/guides/billing-and-subscriptions/) in your account.
+- Access to Azion Edge Storage for storing video content.
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/guides/edge-application/ea-enforce-hls.mdx:26]:
```suggestion
- Access to [Azion Edge Storage](/en/documentation/products/store/edge-storage/) for storing video content.
```

**Review by MarianaAguilera — APPROVED** [review]

File: src/content/docs/pt-br/pages/guias/edge-application/ea-enforce-hls.mdx [hunk 11]
```diff
@@ -14,7 +14,7 @@ import InterfaceNote from '~/includes/snippets/InterfaceNote/pt/snippet.mdx'
 
 <InterfaceNote />
 
-A Plataforma de Edge da Azion permite que você [entregue conteúdo de streaming ao vivo no formato HLS](https://www.azion.com/pt-br/solucoes/live-streaming/), capacitando-lhe a autoprovisionar e configurar suas política de cache facilmente. Este guia abrange o passo a passo para implementar o cache HLS, gerenciando o cache de chunks e playlists e configurando regras no Rules Engine.
+A Plataforma de Edge da Azion permite que você [entregue conteúdo de streaming ao vivo no formato HLS](https://www.azion.com/pt-br/solucoes/live-streaming/), capacitando-lhe a autoprovisionar e configurar suas política de cache facilmente. Este guia abrange o passo a passo para implementar o cache HLS, gerenciar o cache de chunks e playlists, configurar regras no Rules Engine e utilizar o Edge Storage da para uma entrega de vídeo otimizada.
```
Comment (guiafonso-ol) [inline src/content/docs/pt-br/pages/guias/edge-application/ea-enforce-hls.mdx:17]: Talvez incluir o link de Edge Storage?
```suggestion
A Plataforma de Edge da Azion permite que você [entregue conteúdo de streaming ao vivo no formato HLS](https://www.azion.com/pt-br/solucoes/live-streaming/), capacitando-lhe a autoprovisionar e configurar suas política de cache facilmente. Este guia abrange o passo a passo para implementar o cache HLS, gerenciar o cache de chunks e playlists, configurar regras no Rules Engine e utilizar o Edge Storage para uma entrega de vídeo otimizada.
```

**Review by guiafonso-ol — APPROVED** [review]

File: src/content/docs/pt-br/pages/guias/edge-application/ea-enforce-hls.mdx (hunk 11 above)
Comment (isidrohernandezazion) [inline src/content/docs/pt-br/pages/guias/edge-application/ea-enforce-hls.mdx]:
```suggestion
A Plataforma de Edge da Azion permite que você [entregue conteúdo de streaming ao vivo no formato HLS](https://www.azion.com/pt-br/solucoes/live-streaming/), capacitando-lhe a autoprovisionar e configurar suas política de cache facilmente. Este guia abrange o passo a passo para implementar o cache HLS, gerenciar o cache de chunks e playlists, configurar regras no Rules Engine e utilizar o Edge Storage para uma entrega de vídeo otimizada.
```

**Review by isidrohernandezazion — APPROVED** [review]

**Pull request merged**